### ISSUE TITLE
Gradio loading animation slows generations down by 10%

### ISSUE THREAD

**Issue by TheEnhas** [basujindal/stable-diffusion#81]
https://github.com/hlky/stable-diffusion-webui#Loading

Also, is there any way to integrate that UI into this fork? Mostly because it (kind of) has inpainting support in img2img, which is much different than the rudimentary support SD has and doesn't need a 3.1 GB latent diffusion checkpoint to work.

**Comment by basujindal** [comment]
I will look into the performance degradation part. But can you tell more about the inpainting support? Does it only use the original Stable diffusion weights for inpainting?

**Comment by TheEnhas** [comment]
I think so, there's no mention of other models to download for that, and it looks like it's done through masking with img2img. I'd try it but I'm kind of afraid of hosing my SD install, I'm lucky I was able to get this fork up and running lol

**Comment by AK391** [comment]
huggingface diffusers supports inpainting might be easy to integrate?

https://github.com/huggingface/diffusers/tree/main/examples/inference#in-painting-using-stable-diffusion

**Comment by pandoras-rabbithole** [comment]
> I will look into the performance degradation part. But can you tell more about the inpainting support? Does it only use the original Stable diffusion weights for inpainting?

I don't think it has actual inpainting support (nor does it claim to).  it just masks the generated portion if img2img which results in generations where the masked portion doesn't line up or match the outside colors and style at all.
